liesel/goose/models.py

```diff
@@ -1,63 +1,37 @@
-"""
-Model interfaces.
-"""
+import warnings
 
-import copy
-from collections.abc import Callable, Sequence
+from .interface import DataclassInterface, DictInterface, LogProbFunction
 
-from ..docs import usedocs
-from .types import ModelInterface, ModelState, Position
 
-LogProbFunction = Callable[[ModelState], float]
-
-
-@usedocs(ModelInterface)
-class DictModel:
+class DictModel(DictInterface):
     """
-    A model interface for a model state represented by a ``dict[str, Array]`` and a
-    corresponding log-probability function.
+    Alias for :class:`.DictInterface`, provided for backwards compatibility.
+
+    .. deprecated:: v0.2.6
+        Use :class:`.DictInterface` instead. This alias will be removed in v0.4.0.
     """
 
     def __init__(self, log_prob_fn: LogProbFunction):
         self._log_prob_fn = log_prob_fn
```

Comment:
We should here just call the init method of super and not reimplement it. Similar for the others.

If you think that makes no sense ignore my comment.

Reply:
I implemented it, I think it's a sensible suggestion. Thanks!